mocktracer/mocktracer.go

```diff
@@ -98,8 +115,21 @@ func (t *MockTracer) Extract(format interface{}, carrier interface{}) (opentraci
 	return extractor.Extract(carrier)
 }
 
-func (t *MockTracer) recordSpan(span *MockSpan) {
+func (t *MockTracer) recordStartedSpan(span *MockSpan) {
+	t.Lock()
+	defer t.Unlock()
+	t.startedSpans = append(t.startedSpans, span)
+}
+
+func (t *MockTracer) recordFinishedSpan(span *MockSpan) {
 	t.Lock()
 	defer t.Unlock()
 	t.finishedSpans = append(t.finishedSpans, span)
+
+	for i := range t.startedSpans {
+		if t.startedSpans[i].SpanContext.SpanID == span.SpanContext.SpanID {
+			t.startedSpans = append(t.startedSpans[:i], t.startedSpans[i+1:]...)
```

Comment:
`t.startedSpans[i+1:]` - I don't remember exactly if this could cause an overflow in Go, but you're potentially using index out of bounds

Reply:
I think its safe, tried it out in a playground here https://play.golang.org/p/TTKWlI51fTF

Reply:
interesting, that same example fails with `[7:]` but ok with `[6:]` even though both are out of bounds, strictly speaking.